src/functional/ffront/parsing/linear_assign.py

```diff
@@ -0,0 +1,140 @@
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import ast
+import copy
+from typing import Iterator
+
+
+class NodeYielder(ast.NodeTransformer):
+    @classmethod
+    def mutate_ast(cls, node: ast.AST) -> ast.AST:
+        result = list(cls().visit(node))
+        if len(result) != 1:
+            raise ValueError("AST was split or lost during the pass. Use `.visit()` instead.")
+        return result[0]
+
+    def visit(self, node: ast.AST) -> Iterator[ast.AST]:
+        result = super().visit(node)
+        if isinstance(result, ast.AST):
+            yield result
+        else:
+            yield from result
+
+
+class SingleAssignTargetPass(NodeYielder):
+    """
+    Split multi target assignments.
+
+    Requires AST in SSA form (see ``SingleStaticAssignPass``) to yield correct results.
+
+    Example
+    -------
+    Usage:
+
+        import ast
+        from functional.ffront.parsing.parsing import get_ast_from_func
+
+        def foo():
+            a = b = 1
+            return a, b
+
+        print(ast.unparse(
+            SingleStaticAssignPass.mutate_ast(
+                get_ast_from_func(foo)
+            )
+        ))
+
+        # will print out:
+
+        def foo():
+            a = 1
+            b = 1
+
+    """
+
+    def visit_Assign(self, node: ast.Assign) -> Iterator[ast.Assign]:
+        for target in node.targets:
+            new_assign = copy.copy(node)
+            new_assign.targets = [target]
+            yield new_assign
+
+
+class UnpackedAssignPass(NodeYielder):
+    """
+    Explicitly unpack assignments.
+
+    Requires AST in SSA form and assumes only single target assigns, check the following passes.
+
+     * ``SingleStaticAssignPass``
+     * ``SingleAssignTargetPass``
+
+    Example
+    -------
+    Usage:
+
+        import ast
+        from functional.ffront.parsing.parsing import get_ast_from_func
+
+        def foo():
+            a0 = 1
+            b0 = 5
+            a1, b1 = b0, a0
+            return a1, b1
+
+        print(ast.unparse(
+            UnpackAssignPass.mutate_ast(
+                get_ast_from_func(foo)
+            )
+        ))
+
+        # will print out:
+
+        def foo():
+            a0 = 1
+            b0 = 5
+            a1 = (b0, a0)[0]
+            b1 = (b0, a0)[1]
+            return a1, b1
+
+        # which would not have been equivalent had the input AST not been in SSA form.
+    """
+
+    def visit_Assign(self, node: ast.Assign) -> Iterator[ast.Assign]:
+        if len(node.targets) != 1:
+            raise ValueError(
+                "AST contains multi target assigns. Please run `SingleAssignTargetPass` first."
+            )
+        if isinstance(target := node.targets[0], (ast.Tuple, ast.List)):
+            yield from self.unpack_assignment(node, targets=target.elts)
+        else:
+            yield node
+
+    def unpack_assignment(
```

Comment:
Just a question (not important): would it make sense to make this method private just to keep the intended API simpler?

```suggestion
    def _unpack_assignment(
```